### PR TITLE
[Keyboard] archerkeyboard/desire65

### PR DESCRIPTION
**Pull request by akyp**
## Description

Add new keyboard archerkeyboard/desire65

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/archerkeyboard/desire65/config.h
Comment (waffle87) [inline keyboards/archerkeyboard/desire65/config.h:1]: This file should be removed.

File: keyboards/archerkeyboard/desire65/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	),
+
+	[3] = LAYOUT(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                                     KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS
+	)
+
```
Comment (waffle87) [inline keyboards/archerkeyboard/desire65/keymaps/default/keymap.c:39]:
```suggestion
    )
```

**Review by waffle87 — CHANGES_REQUESTED** [review]
Since you're targeting the `develop` branch, please remove the `rules.mk` file and rename the `info.json` to `keyboard.json`.

File: keyboards/archerkeyboard/desire65/keymaps/default/keymap.c (hunk 1 above)
Comment (akyp) [reply keyboards/archerkeyboard/desire65/keymaps/default/keymap.c:39]: Done. Thanks.

File: keyboards/archerkeyboard/desire65/config.h
Comment (dunk2k) [inline keyboards/archerkeyboard/desire65/config.h:1]: please delete this file, as it contains no usable content

File: keyboards/archerkeyboard/desire65/keyboard.json [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        "device_version": "0.0.1",
+        "pid": "0x0002",
+        "vid": "0x0361"
+    },
+    "ws2812": {
+        "driver": "vendor",
+        "pin": "GP26"
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (dunk2k) [inline keyboards/archerkeyboard/desire65/keyboard.json:53]: Add Community Layout support part 1
```suggestion
    "community_layouts": ["65_ansi_blocker_tsangan", "65_ansi_blocker_tsangan_split_bs"],
    "layouts": {
        "LAYOUT_65_ansi_blocker_tsangan_split_bs": {
```

File: keyboards/archerkeyboard/desire65/keyboard.json [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                {"label": "Ctrl", "matrix": [4, 0], "x": 0, "y": 4, "w": 1.5},
+                {"label": "Win", "matrix": [4, 2], "x": 1.5, "y": 4},
+                {"label": "Alt", "matrix": [4, 3], "x": 2.5, "y": 4, "w": 1.5},
+                {"matrix": [4, 7], "x": 4, "y": 4, "w": 7},
+                {"label": "Alt", "matrix": [4, 11], "x": 11, "y": 4, "w": 1.5},
+                {"label": "Left", "matrix": [4, 13], "x": 13, "y": 4},
+                {"label": "Down", "matrix": [4, 14], "x": 14, "y": 4},
+                {"label": "Right", "matrix": [4, 15], "x": 15, "y": 4}
+            ]
+        }
```
Comment (dunk2k) [inline keyboards/archerkeyboard/desire65/keyboard.json:123]: Add Community Layout support part 2
```suggestion
        },
        "LAYOUT_65_ansi_blocker_tsangan": {
            "layout": [
                {"label": "Esc", "matrix": [0, 0], "x": 0, "y": 0},
                {"label": "!", "matrix": [0, 1], "x": 1, "y": 0},
                {"label": "@", "matrix": [0, 2], "x": 2, "y": 0},
                {"label": "#", "matrix": [0, 3], "x": 3, "y": 0},
                {"label": "$", "matrix": [0, 4], "x": 4, "y": 0},
                {"label": "%", "matrix": [0, 5], "x": 5, "y": 0},
                {"label": "^", "matrix": [0, 6], "x": 6, "y": 0},
                {"label": "&", "matrix": [0, 7], "x": 7, "y": 0},
                {"label": "*", "matrix": [0, 8], "x": 8, "y": 0},
                {"label": "(", "matrix": [0, 9], "x": 9, "y": 0},
                {"label": ")", "matrix": [0, 10], "x": 10, "y": 0},
                {"label": "_", "matrix": [0, 11], "x": 11, "y": 0},
                {"label": "+", "matrix": [0, 12], "x": 12, "y": 0},
                {"label": "Backspace", "matrix": [0, 14], "x": 13, "y": 0, "w": 2},
                {"label": "Home", "matrix": [0, 15], "x": 15, "y": 0},
                {"label": "Tab", "matrix": [1, 0], "x": 0, "y": 1, "w": 1.5},
                {"label": "Q", "matrix": [1, 2], "x": 1.5, "y": 1},
                {"label": "W", "matrix": [1, 3], "x": 2.5, "y": 1},
                {"label": "E", "matrix": [1, 4], "x": 3.5, "y": 1},
                {"label": "R", "matrix": [1, 5], "x": 4.5, "y": 1},
                {"label": "T", "matrix": [1, 6], "x": 5.5, "y": 1},
                {"label": "Y", "matrix": [1, 7], "x": 6.5, "y": 1},
                {"label": "U", "matrix": [1, 8], "x": 7.5, "y": 1},
                {"label": "I", "matrix": [1, 9], "x": 8.5, "y": 1},
                {"label": "O", "matrix": [1, 10], "x": 9.5, "y": 1},
                {"label": "P", "matrix": [1, 11], "x": 10.5, "y": 1},
                {"label": "{", "matrix": [1, 12], "x": 11.5, "y": 1},
                {"label": "}", "matrix": [1, 13], "x": 12.5, "y": 1},
                {"label": "|", "matrix": [1, 14], "x": 13.5, "y": 1, "w": 1.5},
                {"label": "End", "matrix": [1, 15], "x": 15, "y": 1},
                {"label": "Caps Lock", "matrix": [2, 0], "x": 0, "y": 2, "w": 1.25},
                {"label": "A", "matrix": [2, 2], "x": 1.75, "y": 2},
                {"label": "S", "matrix": [2, 3], "x": 2.75, "y": 2},
                {"label": "D", "matrix": [2, 4], "x": 3.75, "y": 2},
                {"label": "F", "matrix": [2, 5], "x": 4.75, "y": 2},
                {"label": "G", "matrix": [2, 6], "x": 5.75, "y": 2},
                {"label": "H", "matrix": [2, 7], "x": 6.75, "y": 2},
                {"label": "J", "matrix": [2, 8], "x": 7.75, "y": 2},
                {"label": "K", "matrix": [2, 9], "x": 8.75, "y": 2},
                {"label": "L", "matrix": [2, 10], "x": 9.75, "y": 2},
                {"label": ":", "matrix": [2, 11], "x": 10.75, "y": 2},
                {"label": "\"", "matrix": [2, 12], "x": 11.75, "y": 2},
                {"label": "Enter", "matrix": [2, 14], "x": 12.75, "y": 2, "w": 2.25},
                {"label": "PgUp", "matrix": [2, 15], "x": 15, "y": 2},
                {"label": "Shift", "matrix": [3, 0], "x": 0, "y": 3, "w": 2.25},
                {"label": "Z", "matrix": [3, 2], "x": 2.25, "y": 3},
                {"label": "X", "matrix": [3, 3], "x": 3.25, "y": 3},
                {"label": "C", "matrix": [3, 4], "x": 4.25, "y": 3},
                {"label": "V", "matrix": [3, 5], "x": 5.25, "y": 3},
                {"label": "B", "matrix": [3, 6], "x": 6.25, "y": 3},
                {"label": "N", "matrix": [3, 7], "x": 7.25, "y": 3},
                {"label": "M", "matrix": [3, 8], "x": 8.25, "y": 3},
                {"label": "<", "matrix": [3, 9], "x": 9.25, "y": 3},
                {"label": ">", "matrix": [3, 10], "x": 10.25, "y": 3},
                {"label": "?", "matrix": [3, 11], "x": 11.25, "y": 3},
                {"label": "Shift", "matrix": [3, 13], "x": 12.25, "y": 3, "w": 1.75},
                {"label": "Up", "matrix": [3, 14], "x": 14, "y": 3},
                {"label": "PgDn", "matrix": [3, 15], "x": 15, "y": 3},
                {"label": "Ctrl", "matrix": [4, 0], "x": 0, "y": 4, "w": 1.5},
                {"label": "Win", "matrix": [4, 2], "x": 1.5, "y": 4},
                {"label": "Alt", "matrix": [4, 3], "x": 2.5, "y": 4, "w": 1.5},
                {"matrix": [4, 7], "x": 4, "y": 4, "w": 7},
                {"label": "Alt", "matrix": [4, 11], "x": 11, "y": 4, "w": 1.5},
                {"label": "Left", "matrix": [4, 13], "x": 13, "y": 4},
                {"label": "Down", "matrix": [4, 14], "x": 14, "y": 4},
                {"label": "Right", "matrix": [4, 15], "x": 15, "y": 4}
            ]
        }
```

File: keyboards/archerkeyboard/desire65/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                                     KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS
+	),
+
+	[3] = LAYOUT(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                                     KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS
+	)
```
Comment (dunk2k) [inline keyboards/archerkeyboard/desire65/keymaps/default/keymap.c:38]: Add Community Layout support Part 3
```suggestion
	[0] = LAYOUT_65_ansi_blocker_tsangan_split_bs(
		KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_BSPC, KC_HOME,
		KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_END,
		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGUP,
		KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                   KC_UP,   KC_PGDN,
		KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      MO(1),            KC_LEFT, KC_DOWN, KC_RIGHT
	),

    [1] = LAYOUT_65_ansi_blocker_tsangan_split_bs(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS, KC_DEL,  KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_INS,  KC_TRNS, KC_PSCR, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_CALC, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MPLY,                   KC_VOLU, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                                     KC_TRNS,          KC_MPRV, KC_VOLD, KC_MNXT
    ),
```

File: keyboards/archerkeyboard/desire65/readme.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# archerkeyboard/desire65
+
+A desirable 65% keyboard
+
+* Keyboard Maintainer: [ArcherKeyboard](https://github.com/akyp)
+* Hardware Supported: *Desire65 PCB*
```
Comment (dunk2k) [inline keyboards/archerkeyboard/desire65/readme.md:6]:
```suggestion
* Hardware Supported: Desire65 PCB
```

File: keyboards/archerkeyboard/desire65/readme.md
Comment (dunk2k) [inline keyboards/archerkeyboard/desire65/readme.md:1]: An image of this PCB would be helpful 🙂

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/archerkeyboard/desire65/readme.md
Comment (akyp) [reply keyboards/archerkeyboard/desire65/readme.md:1]: There is a link to the PCB's repo which contains renders and actual pictures.

File: keyboards/archerkeyboard/desire65/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_65_ansi_blocker_tsangan_split_bs(
+		KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_BSPC, KC_HOME,
+		KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_END,
+		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGUP,
+		KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                   KC_UP,   KC_PGDN,
+		KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      MO(1),            KC_LEFT, KC_DOWN, KC_RIGHT
+	),
```
Comment (drashna) [inline keyboards/archerkeyboard/desire65/keymaps/default/keymap.c:14]:
```suggestion
    [0] = LAYOUT_65_ansi_blocker_tsangan_split_bs(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_BSPC, KC_HOME,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_END,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGUP,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                   KC_UP,   KC_PGDN,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      MO(1),            KC_LEFT, KC_DOWN, KC_RIGHT
    ),
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/archerkeyboard/desire65/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                   KC_UP,   KC_PGDN,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      MO(1),            KC_LEFT, KC_DOWN, KC_RIGHT
+    ),
+    [1] = LAYOUT_65_ansi_blocker_tsangan_split_bs(
+        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS, KC_DEL,  KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_INS,  KC_TRNS, KC_PSCR, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_CALC, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MPLY,                   KC_VOLU, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                                     KC_TRNS,          KC_MPRV, KC_VOLD, KC_MNXT
```
Comment (dunk2k) [inline keyboards/archerkeyboard/desire65/keymaps/default/keymap.c:20]: formatting
```suggestion
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_BSPC, KC_HOME,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_END,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,  KC_PGUP,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_PGDN,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      MO(1),   KC_LEFT, KC_DOWN, KC_RIGHT
    ),
    [1] = LAYOUT_65_ansi_blocker_tsangan_split_bs(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, KC_DEL,  _______,
        _______, _______, _______, _______, _______, _______, _______, _______, KC_INS,  _______, KC_PSCR, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
        _______, _______, _______, KC_CALC, _______, _______, _______, _______, _______, _______, _______, KC_MPLY, KC_VOLU, _______,
        _______, _______, _______,                            _______,                                     _______, KC_MPRV, KC_VOLD, KC_MNXT
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by akyp** [comment]
Don't know what went wrong but I'll clean up my repo and local files and try again in the short future. Thanks all.

**Pull request closed without merging**